migrations/20200422144952_structure_project_version_continuation_data.js

```diff
@@ -0,0 +1,156 @@
+  MULTIPLE: 'Multple licence numbers found'
+};
+
+const transform = (version, loggingParams = {}, stringifier) => {
+  if (!version) {
+    return;
+  }
+  const continuationQuestion = version['expiring-yes'];
+  // not a continuation
+  if (!continuationQuestion) {
+    return;
+  }
+
+  // already patched
+  if (version['project-continuation']) {
+    return;
+  }
+
+  const { versionId, projectId, projectStatus, versionStatus } = loggingParams;
+
+  const obj = Value.fromJSON(JSON.parse(continuationQuestion));
+  const text = obj.document.nodes.map(node => node.text.trim()).filter(node => node).join('\n\n');
+
+  const ppl = text.match(LICENCE_NUMBER) || [];
+
+  let date;
+  if (text.match(EXPIRYLONG)) {
+    const matched = text.match(EXPIRYLONG);
+    const day = matched[1];
+    const month = matched[3];
+    const year = matched[7];
+    date = moment(`${day} ${month} ${year}`, 'DD MMM YY');
+  } else if (text.match(EXPIRYLONG2)) {
+    const matched = text.match(EXPIRYLONG2);
+    const day = matched[25];
+    const month = matched[1];
+    const year = matched[29];
+    date = moment(`${day} ${month} ${year}`, 'DD MMM YY');
+  } else if (text.match(EXPIRYSHORT)) {
+    const matched = text.match(EXPIRYSHORT);
+    let day = matched[1];
+    let month = matched[3];
+    const year = matched[6];
+    if (day.length === 1) {
+      day = '0' + day;
+    }
+    if (month.length === 1) {
+      month = '0' + month;
+    }
+    date = moment(`${day} ${month} ${year}`, 'DD MM YY');
+  }
+
+  if (!ppl.length && !date) {
+    console.log(`Cannot parse versionId: ${versionId}`);
+    stringifier && stringifier.write([TYPES.NEITHER, projectId, versionId, projectStatus, versionStatus]);
+    return;
+  }
+
+  if (!ppl.length) {
+    console.log(`Unable to parse licence number, versionId: ${versionId}`);
+    stringifier && stringifier.write([TYPES.LICENCE_NUMBER, projectId, versionId, projectStatus, versionStatus]);
+  }
+
+  if (ppl.length > 1) {
+    console.log(`Multple licence numbers found for versionId: ${versionId}`);
+    stringifier && stringifier.write([TYPES.MULTIPLE, projectId, versionId, projectStatus, versionStatus]);
+  }
+
+  if (!date) {
+    console.log(`Unable to parse expiry date, versionId: ${versionId}`);
+    stringifier && stringifier.write([TYPES.DATE, projectId, versionId, projectStatus, versionStatus]);
+  }
+
+  return {
+    'project-continuation': [
+      {
+        'licence-number': get(ppl, 0, null),
+        'expiry-date': date ? date.format('YYYY-MM-DD') : null
+      }
+    ]
+  }
+};
+
+exports.transform = transform;
+
+exports.up = function(knex) {
+  const stringifier = csv();
+  stringifier.write(['Type', 'Project ID', 'Version ID', 'Project Status', 'Version Status']);
+  return Promise.resolve()
+    .then(() => {
+      return knex('project_versions')
+        .select('project_versions.id')
+        .join('projects', 'project_versions.project_id', 'projects.id')
+        .where({ 'schema_version':  1 });
```

Comment:
We only need to care about the ones which have continued use set to true. We should add that as a condition here to avoid doing unnecessary work.